src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/PackageNameDiagnosticRule.java

```diff
@@ -0,0 +1,35 @@
+package com.azure.tools.apiview.processor.diagnostics.rules;
+
+import com.azure.tools.apiview.processor.diagnostics.DiagnosticRule;
+import com.azure.tools.apiview.processor.model.APIListing;
+import com.azure.tools.apiview.processor.model.Diagnostic;
+import com.github.javaparser.ast.CompilationUnit;
+
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.*;
+
+public class PackageNameDiagnosticRule implements DiagnosticRule {
+
+    @Override
+    public void scan(final CompilationUnit cu, final APIListing listing) {
+        getPackageName(cu).ifPresent(packageName -> {
+            // we need to map the issue to the class id, because package text isn't printed in the APIView output
+            getClassName(cu).map(listing.getKnownTypes()::get).ifPresent(typeId -> {
+                if (!packageName.startsWith("com.azure")) {
```

Comment:
You can use the regex below instead to ensure package names are valid. So, we will not need `if` conditions for each of the special chars.

`^com.azure.[a-z0-9]*(\.[a-z0-9]+)+[0-9a-z]$`

Tested here - https://regex101.com/r/eE4aY9/33

Reply:
Thanks!